frontend/src/components/main/surveys-list.test.tsx

```diff
@@ -69,4 +69,23 @@ describe('Surveys list', () => {
       })
     );
   });
+
+  it('not should show a survey when api return error', async () => {
+    const mockResponse = {
+      mock: true,
+    };
+
+    (httpClient as jest.Mocked<AxiosInstance>).post.mockImplementationOnce(
+      jest.fn((url: string, body: AnswerSurveyCreate) => {
+        return Promise.reject({ response: { data: mockResponse } });
+      })
+    );
+
+    await waitFor(() =>
+      expect(httpClient.post).toHaveBeenCalledWith('api/v1/answers_surveys', {
+        survey_id: 1,
+      })
+    );
+    expect(screen.queryByText('Animals survey')).toBeNull();
```

Comment:
Does make sense to check for some error message? Or does this go outside of unit test scope?

Reply:
@iurynogueira